### PR TITLE
Multilingual CLIP Vector Search - Solution

### PR DESCRIPTION
**Pull request by khasinski**
This is the complete solution fully implementing workshop's feature scope. Enjoy!

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,8 +7,11 @@ gem "propshaft"
 # Use pg as the database for Active Record
 gem "sqlite3"
 gem "sqlite-vec"
+gem "neighbor"
+gem "tokenizers"
```
Comment (khasinski) [inline Gemfile:11]: Pre-defined tokenizers that we can use with a tokenizer parameter file, we use them to build a tokenizer for XLM Roberta CLIP

File: Gemfile [hunk 2]
```diff
@@ -7,8 +7,11 @@ gem "propshaft"
 # Use pg as the database for Active Record
 gem "sqlite3"
 gem "sqlite-vec"
+gem "neighbor"
```
Comment (khasinski) [inline Gemfile:10]: This allows us to run KNN queries, but also formats the embeddings for saving in SQLite

File: app/models/clip/text_embedding.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    tokenizer: Tokenizers.from_pretrained("M-CLIP/XLM-Roberta-Large-Vit-B-32"),
+    model_path: "./multilingual_model/textual.onnx"
+  )
+    @tokenizer = tokenizer
+    @model = OnnxRuntime::Model.new(model_path)
+  end
+
+  def call(text)
+    tensor = tokenizer.encode(text).ids
+    attention_mask = Array.new(tensor.size, 1)
```
Comment (khasinski) [inline app/models/clip/text_embedding.rb:12]: In our case we can just pass ids and attention mask of all ones, but normally you want a fixed size vector (ie. context window) and an attention mask that has ones in places with real ids and zeroes for places with padding tokens.
